File: frontend/src/types/diary.ts
/** Mood options aligned with backend MoodType literal */
export type MoodType = 'happy' | 'calm' | 'tired' | 'sad' | 'energetic'

export interface DiaryResponse {
  id: number
  diary_date: string          // ISO date "YYYY-MM-DD"
  content: string | null
  ai_draft: string | null
  mood: MoodType
  photo_ids: number[]         // cover photo is first
  photo_count: number
  cover_photo_id: number | null
  cover_thumbnail_url: string | null
  created_at: string
  updated_at: string
}

export interface DiaryCalendarItem {
  diary_date: string          // "YYYY-MM-DD"
  mood: MoodType
  summary: string | null      // first 80 chars of content
  cover_thumbnail_url: string | null
  photo_count: number
}

export interface DiaryMonthResponse {
  year: number
  month: number
  entries: DiaryCalendarItem[]
}

export interface DiaryUpsertRequest {
  diary_date: string
  content: string | null
  mood: MoodType
  photo_ids: number[]
  cover_photo_id?: number | null
}

export interface DiaryGenerateDraftRequest {
  diary_date: string
  photo_ids: number[]
  mood: MoodType
}

export interface DiaryGenerateDraftResponse {
  draft: string
}

/** Frontend-only: a calendar cell (either filled or empty) */
export interface CalendarCell {
  day: number                  // 1–31 (0 = padding)
  isPadding: boolean
  diary: DiaryCalendarItem | null
}

/** Mood display config */
export interface MoodConfig {
  name: MoodType
  emoji: string
  label: string
  color: string
}

File: backend/app/schemas/diary.py
"""
Pydantic schemas for the Diary module.
"""
from __future__ import annotations

from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, Field

MoodType = Literal["happy", "calm", "tired", "sad", "energetic"]


# ── Request schemas ───────────────────────────────────────────────────────────

class DiaryUpsertRequest(BaseModel):
    """Create or update a diary entry for a given date."""
    diary_date: date
    content: str | None = None
    mood: MoodType = "calm"
    photo_ids: list[int] = Field(default_factory=list)
    cover_photo_id: int | None = None
    """Explicit cover photo — should be photo_ids[0] when set."""


class DiaryGenerateDraftRequest(BaseModel):
    """Request AI draft generation for a given date."""
    diary_date: date
    photo_ids: list[int] = Field(default_factory=list)
    mood: MoodType = "calm"


class DiaryPolishRequest(BaseModel):
    """Request AI polishing of the user's own draft text."""
    text: str = Field(min_length=10)


# ── Response schemas ──────────────────────────────────────────────────────────

class DiaryPhotoItem(BaseModel):
    """Minimal photo info embedded in a diary response."""
    id: int
    thumbnail_url: str | None = None
    file_name: str

    model_config = {"from_attributes": True}


class DiaryResponse(BaseModel):
    id: int
    diary_date: date
    content: str | None
    ai_draft: str | None
    mood: MoodType
    photo_ids: list[int]
    photo_count: int
    cover_photo_id: int | None = None
    cover_thumbnail_url: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DiaryCalendarItem(BaseModel):
    """Lightweight record for calendar grid rendering (month view)."""
    diary_date: date
    mood: MoodType
    summary: str | None
    """First 80 chars of content — used for hover preview."""
    cover_thumbnail_url: str | None
    photo_count: int


class DiaryMonthResponse(BaseModel):
    year: int
    month: int
    entries: list[DiaryCalendarItem]


class DiaryGenerateDraftResponse(BaseModel):
    draft: str
